gnupg/_meta.py

```diff
@@ -111,16 +111,21 @@ def _find_agent(cls):
             identity = this_process.uids
 
         for proc in psutil.process_iter():
-            # In my system proc.name & proc.is_running are methods
-            if (proc.name() == "gpg-agent") and proc.is_running():
-                log.debug("Found gpg-agent process with pid %d" % proc.pid)
-                if _util._running_windows:
-                    if proc.username() == identity:
-                        ownership_match = True
-                else:
-                    # proc.uids & identity are methods to
-                    if proc.uids() == identity():
-                        ownership_match = True
+            try:
+                # In my system proc.name & proc.is_running are methods
+                if (proc.name() == "gpg-agent") and proc.is_running():
+                    log.debug("Found gpg-agent process with pid %d" % proc.pid)
+                    if _util._running_windows:
+                        if proc.username() == identity:
+                            ownership_match = True
+                    else:
+                        # proc.uids & identity are methods to
+                        if proc.uids() == identity():
+                            ownership_match = True
+            except psutil.Error:
+                # Exception when getting proc info, possibly because the
+                # process is zombie / process no longer exist. Just ignore it.
+                pass
```

Comment:
I think I'd prefer to log when these errors occur, just in case, so that it doesn't silently cause something else to fail in some edge case in the future. I'm happy to make that change myself after merging your patch.